### ISSUE TITLE
 Error when sending message to topic

### ISSUE THREAD

**Issue by danistrebel** [Azure/azure-event-hubs-for-kafka#30]
Description
===========
The producer side of the mirror maker is able to create an event hub in the namespace but fails to publish messages and fails with the following error:

```
[] WARN [Producer clientId=mirror_maker_producer] Error while fetching metadata with correlation id 2 : {test=TOPIC_AUTHORIZATION_FAILED} (org.apache.kafka.clients.NetworkClient)

[] ERROR Error when sending message to topic test with key: 14 bytes, value: 68 bytes with error: (org.apache.kafka.clients.producer.internals.ErrorLoggingCallback)

org.apache.kafka.common.errors.TopicAuthorizationException: Not authorized to access topics: [test]
```

How to reproduce
================
* Created a Standard Tier Event Hub
* Setup the Mirrormaker for Kafka (Version 2.1.0) as described here https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-kafka-mirror-maker-tutorial

Has it worked previously?
=========================
Nope

Checklist
=========
 
 - [x] Verified that port 9093 is not blocked by firewall
 - [x] Verified the namespace is either Standard or Dedicated tier (i.e. it is not Basic tier, which isn't supported)
 - [x] Sample you're having trouble with: `Use Kafka MirrorMaker with Event Hubs for Apache Kafka`
 - [ ] Apache Kafka version: `<REPLACE with e.g., 1.1.0>`
 - [x] Kafka client configuration:
```
bootstrap.servers={EVENTHUBS.FQDN}:9093
client.id=mirror_maker_producer
sasl.mechanism=PLAIN
security.protocol=SASL_SSL
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="$ConnectionString" password="{EVENTHUBS.CONNECTION.STRING}";
```
 - [ ] Namespace and EventHub/topic name
 - [x] Consumer or producer failure `Producer Failure`
 - [ ] If consumer, partition ID, group ID`
 - [x] Timestamp in UTC `Fri, 21 Dec 2018 16:53:48 GMT`
 - [x] Client ID `mirror-maker-producer`
 - [ ] Provide all logs (with debugging enabled if possible) or exception call stack
 - [ ] Standalone repro <REPLACE with e.g., Willing/able to send scenario to repro issue>
 - [x] Operating system: `Ubuntu 18.04 LTS>`
 - [ ] Critical issue

**Comment by basilhariri** [comment]
Hi @danistrebel 

Seems like this should work to me, but @arerlend might have run into this more recently than me. 

When you made your Event Hub namespace, did you make sure to click "Enable Kafka"? The last comment in [this issue](https://github.com/Azure/azure-event-hubs-for-kafka/issues/19) shows how to double-check. 

If so, I would try recreating the EventHub from scratch (delete "test" and recreate in the portal) to make sure nothing's going wrong with auto topic creation

**Comment by arerlend** [comment]
Hmm can you also make sure that it's standard tier and not basic tier?  We return Topic Authorization errors when you're running on basic tier.

**Comment by danistrebel** [comment]
Thanks for getting back. I can confirm the following:
- Tried both autogenerated as well as manually generated Eventhubs.
- Kafka Surface is enabled.
- I am using the standard tier

**Comment by arerlend** [comment]
Does your `sasl.jaas.config` property look something like this?

`sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="$ConnectionString" password="Endpoint=sb://my-namespace.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=12345...wxyz";`
``

**Comment by danistrebel** [comment]
Sorry for the delay.. I was out for the holidays:

Yes, I took the config from here https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-kafka-mirror-maker-tutorial:

```
bootstrap.servers={YOUR.EVENTHUBS.FQDN}:9093
client.id=mirror_maker_producer

#Required for Event Hubs
sasl.mechanism=PLAIN
security.protocol=SASL_SSL
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="$ConnectionString" password="{YOUR.EVENTHUBS.CONNECTION.STRING}";
```

and replaced the password placeholder with the connection string from the Azure portal.

**Comment by arerlend** [comment]
Did you use the namespace level connection string or the event hub level connection string?

**Comment by danistrebel** [comment]
Thank you for your support @arerlend. I finally managed to get it to run.

I did two things so I cannot yet attribute the fix to either one of them individually (Or maybe even both are needed):

1. I renamed my mirrored Kafka topics to be all lowercase
1. Deleted and recreated a new Eventhub namespace.

**Comment by arerlend** [comment]
Interesting - do you remember the source and sink names (w/ uppercase letters if present) of the original topics that were causing problems?

I'm not going to close the issue quite yet if there's a case sensitivity problem that I can repro.

**Comment by vs199vishal** [comment]
Test Data #0 from thread #15                                                                                                                                                 org.apache.kafka.common.errors.TimeoutException: Failed to update metadata after 60000 ms.
--

how do i resolve it

**Comment by basilhariri** [comment]
@vs199vishal please open a new issue with all the information requested from the issue template.

**Comment by arerlend** [comment]
Marking as stale, @danistrebel if you have a chance to respond, that would be great.  Otherwise I'll close the issue in about a week.

**Comment by suvv07** [comment]
I am getting **broker may not be available. (org.apache.kafka.clients.networkclient)** warning while connecting event hub using mirror maker even after taking all the precautions mentioned above.

**Comment by danistrebel** [comment]
Sorry, @arerlend must have slipped. 
The Opic Name was `test2-OrgProductName` where org and product name are camel case.

**Comment by arerlend** [comment]
Closing issue, Kafka topic metadata in EH is not case sensitive.
